dotCMS/src/main/java/com/dotmarketing/business/DeterministicIdentifierAPIImpl.java

```diff
@@ -0,0 +1,374 @@
+        }
+        return name;
+    }
+
+    /**
+     * Generates the seed to the deterministic id
+     * @param contentType
+     * @param contentTypeVarName
+     * @return
+     */
+    private String deterministicIdSeed(final ContentType contentType, final Supplier<String>contentTypeVarName) {
+       final String assetType = resolveAssetType(contentType);
+       final String name = resolveName(contentType, contentTypeVarName);
+       final String deterministicId = String.format("%s:%s",assetType, name).toLowerCase();
+
+       Logger.debug(DeterministicIdentifierAPIImpl.class, String.format(" contentType: %s, assetName: %s,  deterministicId: %s", assetType, name, deterministicId));
+       return deterministicId;
+    }
+
+    /**
+     * Generates the seed to the deterministic id
+     * @param lang
+     * @return
+     */
+    private String deterministicIdSeed(final Language lang){
+       return  String.format("%s:%s:%s", Language.class.getClass().getSimpleName(),lang.getLanguageCode(),lang.getCountryCode());
+    }
+
+    /**
+     * This method is responsible for calling the hashFunction and trim such result down to the max identifier length that guarantees compatibility
+     * @param stringToHash
+     * @return
+     */
+    private String hash(final String stringToHash) {
+        final String hashed = hashFunction.apply(stringToHash).substring(0, 32);
+        Logger.info(DeterministicIdentifierAPIImpl.class,
+                "hashing id: " + stringToHash + " to " + hashed);
+        return hashed;
+    }
+
+    @CloseDBIfOpened
+    private String bestEffortDeterministicId(final String hash) {
+
+        final IdentifierFactory identifierFactory = FactoryLocator.getIdentifierFactory();
+        String candidateId = hash;
+
+        for (int i = 1; i <= MAX_ATTEMPTS; i++) {
+            if (!identifierFactory.isIdentifier(candidateId)) {
+                return candidateId;
+            }
+            candidateId = uuidSupplier.get();
+        }
+        Logger.warn(DeterministicIdentifierAPIImpl.class,String.format("Attempt to generate id has failed instead a non deterministic id (%s) was returned.",candidateId));
+        return candidateId;
+    }
+
+    @CloseDBIfOpened
+    private String bestEffortDeterministicContentId(final String hash) {
+
+        String candidateId = hash;
+
+        for (int i = 1; i <= MAX_ATTEMPTS; i++) {
+            if (!isContentTypeInode(candidateId)) {
+                return candidateId;
+            }
+            candidateId = uuidSupplier.get();
+        }
+        Logger.warn(DeterministicIdentifierAPIImpl.class,String.format("Attempt to generate content-type id has failed instead a non deterministic id (%s) was returned.",candidateId));
+        return candidateId;
+    }
+
+    /**
+     * Test if the calculated hash has already been used
+     * @param hash
+     * @return
+     */
+    private boolean isContentTypeInode(final String hash){
```

Comment:
I guess it needs CloseDBIfOpened as others

Reply:
done. However, these methods are consumed from another which is already marked with CloseDBIfOpened